servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -1632,7 +1641,20 @@ public final Completable flatMapCompletableDelayError(Function<? super T, ? exte
      */
     public final Publisher<T> timeout(long duration, TimeUnit unit,
                                       io.servicetalk.concurrent.Executor timeoutExecutor) {
-        return new TimeoutPublisher<>(this, executor, duration, unit, timeoutExecutor);
+        return new TimeoutPublisher<>(this, executor, duration, unit, true, timeoutExecutor);
+    }
+
+    /**
+     * Converts a {@code Duration} to nanoseconds or if the resulting value would overflow a 64-bit signed integer then
+     * either {@code Long.MIN_VALUE} or {@code Long.MAX_VALUE} as appropriate.
+     *
+     * @param duration The duration to convert
+     * @return The converted nanoseconds value.
+     */
+    private long toNanos(Duration duration) {
```

Comment:
nit: time utility state/methods can be static and likely moved into another class (publisher has many methods so we try to keep it focused on operators).

Reply:
OK. I made this change because I notice the Java 11 has it built-in and looking at the implementation realized that overflow handling was needed.